tests/containers/podman_network.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman network
+# Summary: Test podman network
+# Maintainer: qac team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use version_utils qw(package_version_cmp);
+use containers::utils qw(registry_url container_ip);
+use containers::utils qw(get_podman_version);
+
+sub run() {
+
+    my ($self, $args) = @_;
+    select_serial_terminal;
+    my $podman = $self->containers_factory('podman');
+
+    my $podman_version = get_podman_version();
+
+    record_info('Test', 'Create new networks named newnet1 and newnet2');
+    assert_script_run('podman network create newnet1');
+    assert_script_run('podman network create newnet2');
+
+    record_info('Networks', script_output('podman network ls'));
+    if (package_version_cmp($podman_version, '4.0.0') > 0) {
+        assert_script_run('podman network exists newnet1');
+        assert_script_run('podman network exists newnet2');
+    } else {
+        validate_script_output('podman network ls', sub { m/newnet[0-9]+/g });
+    }
+
+    record_info('Prep', 'Create two more networks named newnet3 and newnet4');
+    assert_script_run('podman network create newnet3');
+    assert_script_run('podman network create newnet4');
+
+    record_info('Test', 'Delete newnet3 and list the networks to see if it is deleted');
+    assert_script_run('podman network rm newnet3');
+    if (package_version_cmp($podman_version, '4.0.0') > 0) {
```

Comment:
suggestion: you could maybe rely on just grepping the output for network existence considering that there'd be a lot more branching as and when more tests are added to this suite.

Reply:
I suggested him to still use `exists` option to cover all possible cases of the `podman network` command.

Reply:
@mgrossu Maybe to avoid repeating this if-else twice in this module, you can create a sub routine in this same file `sub network_exists` and move this code there. Then, in this flow, you would only do:
```
assert_script_run('podman network rm newnet3');
network_exists('newnet3', true);
```
You can use the second parameter to check if the network should exists (true) or not (false).

Reply:
updated the code.